### DISCUSSION TITLE
Week 8 - Don’t trust a statistic you didn’t fake yourself

### DISCUSSION THREAD

**Discussion by WeberWr** [The-Escape-Box/TheEscapeBox#39] [General]
Welcome back!

Today we want to tell you something about analytics and how what we think would be good data we can collect from our players. We also did some stuff in our game like adding the new enemy, trying to work out a minimap (spoiler it doesn't work) and clean up our assets a bit, but we will not tell you more about this now. You we because you see the "finished" game next week.

### The Analytics

We mainly thought that two different categories of data is necessary. On the one hand we have data over the adventure in the game and on the other hand there is data about the fights you have in the adventure. This data can get processed and put in some datastorage.  

<img width="1078" alt="The analytics design" src="https://github.com/The-Escape-Box/TheEscapeBox/assets/97832247/b1b210d4-0923-476b-9b05-0ae72da15852">

The datastorage can look something like the following. This is only an example because we do not really now what kind of storage unity provides because its hidden behind a paywall. 

We also looked at UGS (Unity Game Services) and are sure we should use it, because it just gets you a base analytics(player played the game etc.). If the customization is good enough we can push all of our data there to get the best analytics. Together with the Unity SQL Data Explorer we think its the important thing is, that we have a good data base and collect all data we need.
 
<img width="190" alt="SQL" src="https://github.com/The-Escape-Box/TheEscapeBox/assets/97832247/6a5dddbb-f1c7-452c-890e-a397aafc3269">

Besides from the data we collect while you playing we could open our discord server for all players to let you talk direct with us and post your opinion over the game. So what brings data if we can't interpret it, right?

## Time spend
Time we spent this week:
Mohamad: 8h
Lukas: 8h

We hope to see you next week so that we all can bring the journey of this game to an end.

**Comment by jonasNuber** [comment]
Hey there EscapeBox! 
I am exited to see the new features you added during the last week later, esspecially the minimap, which was a feature I really hoped would be added along the line. I am not entirely sure how it will inpact the gameplay, but I think your game should be able to benefit from it instead of it taking away from it. It's nice to see, that you are still motivated to add new stuff anyhow :).

Your analytic categories seem sensible. They give a good overview of how well a player is able to perform during a playthrough, which also could enable the possibility for statistics. Which of course the players can directly benefit from, but also could be used to identify bottlenecks in the game, where the players take longer than they should and also find stages which might be too easy. I think the database approach is fine, but how would it work when the game is actually released and played by players all over? When it is just on your machine you can directly push it to the data store, but for remote use, would you plan to create a web endpoint statistcs can be send to automatically from the clients, or would you go for a more manual approach?
The idea to host a public discord server i highly approve of. Automatic anonymised quantative data is a pretty usuful resource to make your game better, but direct feedback from players and maybe also being able to source their ideas and creativity is always a good way to improve the gameplay and maybe think about your software in a way you yourself wouldn't have been able to.

Anyway, keep up the good work and stay positive^^!
